### ISSUE TITLE
Command 'Restart the clangd language server' resulted in an error (command 'clangd.restart' not found)

### ISSUE THREAD

**Issue by studgeek** [clangd/vscode-clangd#94]
When I run the command 'Restart the clangd language server' (added in #35) I get the following error. It has worked in the past.
```
Command 'Restart the clangd language server' resulted in an error (command 'clangd.restart' not found)
```

**Logs**
I don't see any clangd logs in Output window.                     

**System information**
Clangd version (from the log, or `clangd --version`): llvm-trunk 0581c0b0eeba03da590d1176a4580cf9b9e8d1e3
clangd extension version: v0.1.7
Operating system: linux

**Comment by studgeek** [comment]
I don't see an error when I just reload the window.

**Comment by HighCommander4** [comment]
If you open the vscode devtools (`Help --> Toggle Developer Tools`), are there any other errors in the `Console` tab?

**Comment by studgeek** [comment]
I since restarted both the client and the workstation (I am using SSH remote) and it seems to no longer have the problem. Will re-open if I see it again.
Thanks for tip on using Toggle Developer Tools, will use it if see issue in future.

**Comment by milesfrain** [comment]
I'm able to consistently reproduce this issue if I attempt to `Restart the clangd language server` when clangd has not already be started (and therefore `clangd.restart` has not yet been registered).

This is a bit of an unusual workflow, but here's an example of how it can happen:
- Open a cpp project in vscode
- Open a non-cpp file (e.g. a bazel `BUILD` file)
- Close vscode
- Open vscode
  - The non-cpp file will be remembered as opened and not trigger the clangd extension to automatically start.
- Run the `Restart the clangd language server`, which triggers error.
  - Motivation for running the restart command could be that I just regenerated a new `compile_commands.json` file due to swapping to a different branch or target after launching vscode and assumed clangd was already running.

Two ways to suppress this error are to either:
- Open a .cpp file to automatically start the extension
- Run the `Manually activate clangd extension` command

This error is simple enough to work-around/ignore once you know the reason for the it, but ux could be improved to spare folks from diving into a rabbit hole. For example, maybe all the clangd commands should be registered to either automatically start the extension or show a better error message.

 cc @hokein (original author of #35).

**Comment by HighCommander4** [comment]
> For example, maybe all the clangd commands should be registered to either automatically start the extension

VSCode started doing this automatically in 1.74: https://code.visualstudio.com/updates/v1_74#_implicit-activation-events-for-declared-extension-contributions

However, there was a bug in vscode-clangd that caused the `clangd.restart` command's handler to fail with another error when invoked in this situation. That was fixed in #502.
